pkg/splunk/enterprise/util.go

```diff
@@ -33,10 +34,69 @@ import (
 // kubernetes logger used by splunk.enterprise package
 var log = logf.Log.WithName("splunk.enterprise")
 
+// ApplyCommonSecretObject creates/updates the namespace scoped "splunk-secrets" K8S secret object
+func ApplyCommonSecretObject(client splcommon.ControllerClient, namespace string) (*corev1.Secret, error) {
+	var current corev1.Secret
+
+	// Types of Splunk secret tokens
+	secretTokenTypes := []string{"hec_token", "password", "pass4symmkey", "idxc_secret", "shc_secret"}
+
+	// Check if a K8S secrets object "splunk-secrets" exists in the namespace
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: commonSecretName}
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err != nil {
+		// Not found, generate values for all types of tokens
+		secretData := make(map[string][]byte)
+		for _, tokenType := range secretTokenTypes {
+			secretData[tokenType] = splcommon.GenerateSecret(secretBytes, 24)
```

Comment:
Note that hec token is currently handled as a special case (formatting)

Reply:
Fixed in the new commit 9a807f8cbe5f48b760a5123c35f5816d437f1752